src/newgrf_cargo.cpp

```diff
@@ -30,6 +35,16 @@ struct CargoResolverObject : public ResolverObject {
 	return nullptr;
 }
 
+GrfSpecFeature CargoResolverObject::GetFeature() const
+{
+	return GSF_CARGOES;
+}
+
+uint32 CargoResolverObject::GetLocalID() const
+{
+	return this->cargospec->label; // Not quite the same as ID, but should work
```

Comment:
Technically the local id is "cargospec->Index()". But I agree that cargoslots and cargobits are confusing and "label" is more useful.
However, the method should not be named "GetLocalID" then.